src/libraries/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -789,6 +789,14 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { (567.89m).ToString(), defaultStyle, null, 567.89m };
             yield return new object[] { (-567.89m).ToString(), defaultStyle, null, -567.89m };
 
+            yield return new object[] { ".234", defaultStyle, null, 0.234m };
+            yield return new object[] { "9813.", defaultStyle, null, 9813.0m };
+
+            string inputForDecimalWithDotAtMaxDigit = "6" + new string('1', 27) + ".";
+            yield return new object[] { inputForDecimalWithDotAtMaxDigit, defaultStyle, null, Decimal.Parse(inputForDecimalWithDotAtMaxDigit) };
```

Comment:
I think you should replace `Decimal.Parse(inputForDecimalWithDotAtMaxDigit)` with `61111...111.m`. That is because this unit test is called by the `Parse` method here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System/DecimalTests.cs#L814. If you look at the implementation of that method, it calls `decimal.TryParse(value)` and check if the returned `out result` is equal to `expected` on the next line (Line 824). With

```csharp
yield return new object[] { inputForDecimalWithDotAtMaxDigit, defaultStyle, null, Decimal.Parse(inputForDecimalWithDotAtMaxDigit) };
```
 `value` will be inputForDecimalWithDotAtMaxDigit and `expected` will be `Decimal.Parse(inputForDecimalWithDotAtMaxDigit)`. So, we will end up checking `Assert.Equal(Decimal.Parse(inputForDecimalWithDotAtMaxDigit), Decimal.Parse(inputForDecimalWithDotAtMaxDigit))` which means that we are not really testing `result` with a known correct `expected` value. So, the way to fix this is to specify `expected` directly as a decimal value.

Similar comments for double, single and half!